operator-pipeline-images/operatorcert/entrypoints/request_signature.py

```diff
@@ -0,0 +1,248 @@
+        required=True,
+    )
+    parser.add_argument(
+        "--sig-key-id",
+        default="4096R/55A34A82 SHA-256",
+        help="The signing key id that the content was signed with",
+        required=True,
+    )
+    parser.add_argument(
+        "--sig-key-name",
+        default="containerisvsign",
+        help="The signing key name that the content was signed with",
+        required=True,
+    )
+    parser.add_argument(
+        "--umb-client-name",
+        default="operatorpipelines",
+        help="Client name to connect to umb, usually a service account name",
+        required=True,
+    )
+    parser.add_argument(
+        "--umb-listen-topic",
+        default="VirtualTopic.eng.robosignatory.isv.sign",
+        help="umb topic to listen to for responses with signed content",
+        required=True,
+    )
+    parser.add_argument(
+        "--umb-publish-topic",
+        default="VirtualTopic.eng.operatorpipelines.isv.sign",
+        help="umb topic to publish to for requesting signing",
+        required=True,
+    )
+    parser.add_argument(
+        "--umb-url",
+        default="umb.api.redhat.com",
+        help="umb host to connect to for messaging",
+        required=True,
+    )
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+global umb
+umb = None
+
+global result_file
+result_file = None
+
+
+class UmbHandler(stomp.ConnectionListener):
+    def on_error(self, frame):
+        logging.error("Received an error frame:\n{}".format(frame.body))
+
+    def on_message(self, frame):
+        # handle response from radas
+        t = threading.Thread(target=process_message, args=[frame])
+        t.start()
+
+    def on_disconnected(self):
+        logging.info("Disconnected from umb.")
+
+
+def start_umb_client(hosts: list, client_name: str):
+    # start the message bus listener
+    host_list = []
+
+    for host in hosts:
+        host_list.append((host, 61612))
+
+    cert = os.environ.get("UMB_CERT_PATH")
+    key = os.environ.get("UMB_KEY_PATH")
+
+    if cert and key:
+        if os.path.exists(cert) and os.path.exists(key):
+            LOGGER.debug("umb listener session with cert + key is created.")
+
+            umb_client = UmbClient(
+                host_list,
+                os.getenv("UMB_CERT_PATH"),
+                os.getenv("UMB_KEY_PATH"),
+                UmbHandler(),
+                client_name,
+            )
+        else:
+            LOGGER.error(
+                "UMB_CERT_PATH or UMB_KEY_PATH does not point to a file that exists"
+            )
+            sys.exit(1)
+    else:
+        LOGGER.error(
+            "No auth details provided for umb. Define UMB_CERT_PATH + UMB_KEY_PATH."
+        )
+        sys.exit(1)
+
+    return umb_client
+
+
+def process_message(frame: Any):
+    msg_dict = json.loads(frame.body)
+    msg = msg_dict["msg"]
+
+    global umb
+    if msg.get("request_id") == umb.id:
```

Comment:
Do I understand it right that the task sends a signature request with `id` field and waits for responses with the same `request_id` to match a request with a response?

Reply:
Yes, the `request_id` is how we can tell which signature is for the request we sent, when there are multiple going on.